stl/inc/typeindex

```diff
@@ -38,9 +38,17 @@ public:
         return *_Tptr == *_Right._Tptr;
     }
 
+#if !_HAS_CXX20
     _NODISCARD bool operator!=(const type_index& _Right) const noexcept {
         return !(*this == _Right);
     }
+#else // ^^^ !_HAS_CXX20 / _HAS_CXX20 vvv
+    _NODISCARD strong_ordering operator<=>(const type_index& _Right) const noexcept {
+        return *_Tptr == *_Right._Tptr      ? strong_ordering::equal
+             : _Tptr->before(*_Right._Tptr) ? strong_ordering::less
+                                            : strong_ordering::greater;
+    }
+#endif // !_HAS_CXX20

```

Comment:
You should remove the relational operators for C++20 i think

Reply:
I think not for typeindex. Im seeing it [here](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2019/p1614r2.html#clause-20-general-utilities-library) under Change 20.17.2 [type.index.overview].

Reply:
> I think not for typeindex. Im seeing it [here](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2019/p1614r2.html#clause-20-general-utilities-library) under Change 20.17.2 [type.index.overview].

The working draft agrees. `type_index` is weird because `type_info` has no `<=>`, and it's basically a handle to an instance of `type_info`.